test/oqs_test_tlssig.c

```diff
@@ -0,0 +1,188 @@
+  int ret = 1, testresult = 0;
+  char certpath[300];
+  char privkeypath[300];
+# ifndef OPENSSL_SYS_VMS
+    const char *sep = "/";
+# else
+    const char *sep = "";
+# endif
+
+  if (!alg_is_enabled(sig_name)) {
+     printf("Not testing disabled algorithm %s.\n", sig_name);
+     return 1;
+  }
+
+  sprintf(certpath, "%s%s%s%s", certsdir, sep, sig_name, "_srv.crt");
+  sprintf(privkeypath, "%s%s%s%s", certsdir, sep, sig_name, "_srv.key");
+  testresult =
+    create_ssl_ctx_pair(libctx, TLS_server_method(), TLS_client_method(),
+                        TLS1_3_VERSION, TLS1_3_VERSION,
+                        &sctx, &cctx, certpath, privkeypath);
+
+  if (!testresult) {
+      ret = -1; goto err;
+  }
+
+  testresult =
+    create_ssl_objects(sctx, cctx, &serverssl, &clientssl, NULL, NULL);
+
+  if (!testresult) {
+      ret = -2; goto err;
+  }
+
+  testresult =
+    create_ssl_connection(serverssl, clientssl, SSL_ERROR_NONE);
+  if (!testresult) {
+      ret = -5; goto err;
+  }
+
+  err:
+  SSL_free(serverssl);
+  SSL_free(clientssl);
+  SSL_CTX_free(sctx);
+  SSL_CTX_free(cctx);
+  return ret;
+}
+
+static void test_oqs_sigs(EVP_SIGNATURE *evpsig, void *vp) {
+	OSSL_PROVIDER* prov = EVP_SIGNATURE_get0_provider(evpsig);
+	if (!strcmp(OSSL_PROVIDER_get0_name(prov), "oqsprovider")) {
+		printf("Commencing test of %s:\n", EVP_SIGNATURE_get0_name(evpsig));
+		test_oqs_tlssig(EVP_SIGNATURE_get0_name(evpsig)); 
+	}
+}
+
+static int test_signature(const OSSL_PARAM params[], void *data)
+{
+    int ret = 0;
+    int *errcnt = (int *) data;
+    // Change define over to OSSL... once upstream update is done
+    const OSSL_PARAM *p = OSSL_PARAM_locate_const(params, OQS_CAPABILITY_TLS_SIGALG_NAME);
+    if (p == NULL || p->data_type != OSSL_PARAM_UTF8_STRING) {
+        ret = -1;
+        goto err;
+    }
+
+    char* sigalg_name = OPENSSL_strdup(p->data);
+
+    if (sigalg_name == NULL) return 0;
+
+    fprintf(stderr,
+            cGREEN "  Testing...: %s" cNORM "\n",
+            sigalg_name);
+
+    ret = test_oqs_tlssig(sigalg_name);
+
+    if (ret >= 0) {
+        fprintf(stderr,
+                cGREEN "  TLS-SIG handshake test succeeded: %s" cNORM "\n",
+                sigalg_name);
+    } else {
+        fprintf(stderr,
+                cRED "  TLS-SIG handshake test failed: %s, return code: %d"  cNORM "\n",
+                sigalg_name, ret);
+        ERR_print_errors_fp(stderr);
+        (*errcnt)++;
+    }
+
+    err:
+    OPENSSL_free(sigalg_name);
+    return ret;
+}
+
+static int test_provider_signatures(OSSL_PROVIDER *provider, void *vctx)
+{
+    const char* provname = OSSL_PROVIDER_get0_name(provider);
+
+    if (!strcmp(provname, PROVIDER_NAME_OQS))
+        return OSSL_PROVIDER_get_capabilities(provider, "TLS-SIGALG",
+                                              test_signature, vctx);
+    else
+        return 1;
+}
+
+#define nelem(a) (sizeof(a)/sizeof((a)[0]))
+
+static size_t tcb(const char *buf, size_t cnt,
```

Comment:
Function seems to be unused with the commented-out section in main(). Maybe add a ifdef?